plugins/org.locationtech.udig.core/src/org/locationtech/udig/core/internal/FeatureUtils.java

```diff
@@ -333,12 +338,36 @@ public int size() {
     private static Object[] copyAttributes( SimpleFeatureType destSchema, SimpleFeature source,
             Map<String, Iterator< ? extends Geometry>> geometries, Map<String, String> attributeMap, MathTransform mt  ) {
         Object[] attributes = new Object[destSchema.getAttributeCount()];
+        
+        //dummy progress monitor
+        IProgressMonitor mon = new NullProgressMonitor();
+        
         for( int i = 0; i < attributes.length; i++ ) {
             String sourceAttributeName = destSchema.getDescriptor(i).getName().getLocalPart();
             String name = attributeMap.get(sourceAttributeName);
-            if (name != null)
-                attributes[i] = source.getAttribute(name);
-            else {
+            //if an attribute match exists set its value
+            if (name != null) {                 
+                if (destSchema.getDescriptor(sourceAttributeName).getType().getBinding().isAssignableFrom(
+                        source.getFeatureType().getDescriptor(sourceAttributeName).getType().getBinding())) {
+                    attributes[i] = source.getAttribute(name);
+                } else {
+                    //if not directly assignable then check for an adapter
+                    if (AdapterUtil.instance.canAdaptTo(
+                            source.getAttribute(sourceAttributeName), 
+                            destSchema.getDescriptor(i).getType().getBinding())) {
+                        try {
+                            attributes[i] = AdapterUtil.instance.adaptTo(
+                                    destSchema.getDescriptor(i).getType().getBinding(), 
+                                    source.getAttribute(sourceAttributeName), mon);
+                        } catch (Exception e) {
+                            e.printStackTrace();
```

Comment:
can we log it to Plugin-Logger rather than just print it?

Reply:
done